crates/ruma-common/src/events/receipt.rs

```diff
@@ -102,13 +104,149 @@ pub struct Receipt {
     /// The time when the receipt was sent.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub ts: Option<MilliSecondsSinceUnixEpoch>,
+
+    /// The thread this receipt applies to.
+    #[serde(rename = "thread_id", skip_serializing_if = "crate::serde::is_default")]
```

Comment:
Have you tested it? I'm very certain it doesn't work because `Option` deserializing fine from an omitted field is not a feature of its `Deserialize` implementation but rather a special-casing in the derive macros.

Reply:
Well yes, there is a test with a `Receipt` with no fields. However now that the deserialization changed, we deserialize from a `Cow<str>` so it's needed.